docs/src/types/vectors.md

```diff
@@ -0,0 +1,183 @@
+# Vectors
+
+## Passing in vectors
+You can pass a Rust std::vec::Vec into your contract function transparently. The following code calls a sway function which accepts and then returns a `Vec<SomeStruct<u32>>`.
+
+```rust,ignore
+{{#include ../../../packages/fuels/tests/harness.rs:passing_in_vec}}
+```
+
+You can use a vector just like you would use any other type -- i.e. a `[Vec<u32>; 2]` or a `SomeStruct<Vec<Bits256>>` etc.
+
+## Returning vectors
+There is a mandatory extra step to returning vectors -- you need to `log` every element before returning the vector.
+
+You must not mix any unrelated `log`s once you start logging vector elements. 
+
+These logs must be the last ones you make.
+
+
+### A simple case:
+```Rust
+contract;
```

Comment:
Can we include actual code from the test directory instead of pasting code here?

Reply:
Sure. Wanted to originally but those were a bit convoluted with assertion checks for equality and what not.

I'll add some that will serve mostly as an example and link to them here.